nengo/builder/learning_rules.py

```diff
@@ -278,6 +371,253 @@ def step_simvoja():
+    error : Signal
+        The error driving the learning, :math:`E`
+    n_neurons : int
+        The number of neurons in the ensemble, :math:`n`
+    apply_every : float or None
+        A scalar indicating how often to apply learning rule.
+    n_steps : Signal
+        The integer timestep which the simulator is executing.
+    tag : str, optional (Default: None)
+        A label associated with the operator, for debugging purposes.
+
+    Attributes
+    ----------
+    apply_every : float or None
+        A scalar indicating how often to apply learning rule.
+    correction : Signal
+        The scaled negative error
+    delta : Signal
+        The synaptic weight change to be applied, :math:`\Delta \omega_{ij}`.
+    error : Signal
+        The error driving the learning, :math:`E`
+    learning_rate : float
+        The scalar learning rate, :math:`\kappa`.
+    n_neurons : int
+        The number of neurons in the ensemble, :math:`n`
+    n_steps : Signal
+        The integer timestep which the simulator is executing.
+    pre_filtered : Signal
+        The presynaptic activity, :math:`a_i`.
+    tag : str or None
+        A label associated with the operator, for debugging purposes.
+
+    Notes
+    -----
+    1. sets ``[]``
+    2. incs ``[]``
+    3. reads ``[pre_filtered, error, n_steps]``
+    4. updates ``[delta, correction]``
+    """
+
+    def __init__(self, pre_filtered, delta, correction, learning_rate, error,
+                 n_neurons, apply_every, n_steps, tag=None):
+        super(SimPESDecoders, self).__init__(tag=tag)
+        self.pre_filtered = pre_filtered
+        self.delta = delta
+        self.correction = correction
+        self.learning_rate = learning_rate
+        self.error = error
+        self.n_neurons = n_neurons
+        self.n_steps = n_steps
+        self.apply_every = apply_every
+
+        self.sets = []
+        self.incs = []
+        self.reads = [pre_filtered, error, n_steps]
+        self.updates = [delta, correction]
+
+    def _descstr(self):
+        return 'pre=%s, err=%s -> %s' % (
+            self.pre_filtered, self.error, self.delta)
+
+    def make_step(self, signals, dt, rng):
+        pre_filtered = signals[self.pre_filtered]
+        delta = signals[self.delta]
+        correction = signals[self.correction]
+        alpha = self.learning_rate * dt
+        n_neurons = self.n_neurons
+        error = signals[self.error]
+        period = (1 if self.apply_every is None else
+                  self.apply_every / dt)
+
+        if period > 1:
+            n_steps = signals[self.n_steps]
+
+            def step_simpesdecoders_sometimes():
+                if n_steps % period < 1:
+                    correction[...] = -alpha / n_neurons * error
+                    delta[...] = np.outer(correction, pre_filtered)
+
+                    # scale to compensate
+                    delta[...] *= period
+
+                else:
+                    delta[...] = 0
```

Comment:
Do we want to set `correction[...] = 0` here also? That will make probing more intuitive as it holds `0` instead of the previous value. Is it worth the extra step?